app/app.js

```diff
@@ -120,15 +125,143 @@ let Marvel = {
         showFirstLast: false
       })
     );
+  },
+  addRefinementList(){
+    this.search.addWidget(
+      instantsearch.widgets.currentRefinedValues({
+        container: '#current-refined-values',
+        clearAll: 'after'
+      })
+    );
+  },
+  renderHits(){
+    var renderedHits = {
+      render: function(options) {
+        processHeroProfile();
+      }
+    };
+
+    this.search.addWidget(renderedHits);
   }
+
 };
 
 export default Marvel;
 
+function processHeroProfile(){
+
+  var hit = $('.btn-profile'),
+      results = $('.hits'),
+      profile = $('.hero-profile'),
+      profileName = $('.hero-profile .hero-name'),
+      profileRealName = $('.hero-profile .hero-secret-identity'),
+      profileAvatar = $('.hero-profile .hero-avatar img'),
+      profileDescription = $('.hero-profile .hero-description'),
+      profilePartners = $('.hero-profile .hero-partners'),
+      profileHeroVillain = $('.hero-profile .hero-statement'),
+      profilePowers = $('.hero-profile .hero-powers');
+  
+  hit.each(function(){
+    $(this).click(function(e,t){
+      var $this = $(this);
+      var datas = $(this).closest('.hit').find('.dump').val(),
+          datas = $.parseJSON(datas);
+      
+      // Fetch & define values
+      var
+      heroName = datas.name,
+      heroSecretId = datas.secretIdentities,
+      heroAvatar = datas.image.thumnail,
+      heroBanner = datas.image.banner,
+      heroBackground = datas.image.background,
+      heroDesc = datas.description,
+      heroPowers = datas.powers,
+      heroPartners = datas.partners,
+      isHero = datas.isHero,
+      isVillain = datas.isVillain;
+
+     
+      // Appy them
+      profileName.html(heroName);
+      profileAvatar.attr('src', heroAvatar);
+      profileDescription.html('<p>'+heroDesc+'</p>');
+      profileRealName.html(heroSecretId)
 
+      // Check if he is a hero, a vilain, both, or null
+      if(isHero==true && isVillain==false) {
```

Comment:
That's actually true, I have to remind it, I always do like this, and I already got this kind of comments ^^! Changing it right now!